src/main/java/seedu/duke/commands/DeleteCommand.java

```diff
@@ -1,60 +1,144 @@
 package seedu.duke.commands;
 
 import seedu.duke.Ui;
-import seedu.duke.items.Item;
+import seedu.duke.exceptions.DukeException;
+
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
-import java.util.ArrayList;
-import java.util.Arrays;
 import static seedu.duke.Duke.eventCatalog;
-import static seedu.duke.Duke.taskList;
+
 
 public class DeleteCommand extends Command {
 
-    protected static final String TASK_OR_EVENT_FLAG = "-";
+    protected static final String EVENT_FLAG = "-e";
+    protected static final String TASK_FLAG = "-t";
+    protected static final String DELETE_ALL_FLAG = "all";
 
     // input from user
-    public String taskToDelete;
-    public static ArrayList<Item> combinedItemList = new ArrayList<>();
+    private String itemFlag;
+    private int indexToDelete;
+
+    private boolean isDeleteAll;
+    private boolean isCorrectFormat;
+
+    // create logger
+    private static final Logger logger = Logger.getLogger("Logger");
 
-    // v1.0: deleteCommand deletes purely based on index, i.e. delete [TASK_INDEX]
-    // converted command array to string for future uses, where input may have extra spaces
-    // such as 'delete foo bar'
-    public DeleteCommand(String[] itemDescription) {
-        taskToDelete = Arrays.toString(itemDescription);
+
+    // v2.0: deleteCommand deletes purely based on index, i.e. delete -t/-e [TASK_INDEX]
+    public DeleteCommand(String[] command) {
+
+        isCorrectFormat = true;
```

Comment:
Since an exception is thrown anyways for the incorrect format, is this variable still required?